lib/elasticsearch/config.js

```diff
@@ -0,0 +1,100 @@
+// Configure search scopes:
+const SEARCH_SCOPES = {
+  all: {
+    fields: [
+      'title^5',
+      'title.folded^2',
+      'description.folded',
+      'subjectLiteral^2',
+      'subjectLiteral.folded',
+      'creatorLiteral^2',
+      'creatorLiteral.folded',
+      'contributorLiteral.folded',
+      'note.label.folded',
+      'publisherLiteral.folded',
+      'seriesStatement.folded',
+      'titleAlt.folded',
+      'titleDisplay.folded',
+      'contentsTitle.folded',
+      'donor.folded',
+      'parallelTitle.folded^5',
+      'parallelTitleDisplay.folded',
+      'parallelTitleAlt.folded',
+      'parallelSeriesStatement.folded',
+      'parallelCreatorLiteral.folded',
+      'parallelPublisher',
+      'uniformTitle.folded',
+      'parallelUniformTitle',
+      'formerTitle',
+      'addedAuthorTitle',
+      'placeOfPublication',
+      { field: 'items.idBarcode', on: (q) => /\d{6,}/.test(q) },
+      // Try to detect shelfmark searches (e.g. JFD 16-5143)
+      { field: 'items.shelfMark', on: (q) => /^[A-Z]{1,3} \d{2,}/.test(q) }
+    ]
+  },
+  title: {
+    fields: [
+      'title^5',
+      'title.folded^2',
+      'titleAlt.folded',
+      'uniformTitle.folded',
+      'titleDisplay.folded',
+      'seriesStatement.folded',
+      'contentsTitle.folded',
+      'donor.folded',
+      'parallelTitle.folded^5',
```

Comment:
We specially boost non-folded title just to give latin-based text matches with diacritics an edge over similar matches that match with diacritics removed. The "folding_analyzer" analyzer we're using uses the icu tokenizer and folder. I'm not sure what folding the folder does on non-latin languages, but the tokenizer is key as it knows how to break up cjk, etc into appropriate terms. So until we find otherwise, I think it's sufficient to match parallel values on just the .folded field